### PR TITLE
Add different default messages for different constraints

### PR DESCRIPTION
**Pull request by s1mahanty**
## Related Issue
[FORMS-2981](https://jira.corp.adobe.com/browse/FORMS-2981)

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: ui.frontend/src/view/FormFieldBase.js [hunk 1]
```diff
@@ -221,9 +222,72 @@ export default class FormFieldBase extends FormField {
     updateErrorMessage(errorMessage, state) {
         if (this.errorDiv) {
           this.errorDiv.innerHTML = state.errorMessage;
-          if (state.valid === false && !state.errorMessage) {
-            this.errorDiv.innerHTML = 'There is an error in the field';
-          }
+            if (state.valid === false && !state.errorMessage) {
+                let validationMessage = LanguageUtils.getTranslatedString(this.formContainer.getModel()._jsonModel.lang, "DefaultError");
```
Comment (rismehta) [inline ui.frontend/src/view/FormFieldBase.js:226]: We have expose .lang so please use it directly `this.formContainer.getModel().lang`. Also, we should return a function, to make this more readable. Please check this, https://github.com/adobe/aem-core-forms-components/blob/master/ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/af-commons/v1/clientlibs/editor/utils/utils.js#L104-L124

Comment (rismehta) [reply ui.frontend/src/view/FormFieldBase.js:226]: Usage would be the following,
`
let getTranslatedString = LanguageUtils.getTranslatedString(this.formContainer.getModel().lang);
// create constants for all validationState
// please make the code more readable
if (ValidationStateConstants.VALUE_MISSING in validationState) {
    getTranslatedString(`${ValidationStateConstants.VALUE_MISSING}DefaultError`)
}
`

**Comment by rismehta** [comment]
Please add test cases for this use-case

**Comment by rismehta** [comment]
Please add test cases

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/af-clientlibs/core-forms-components-runtime-all/resources/i18n/de.json [hunk 2]
```diff
@@ -1,7 +1,24 @@
...
-  "FileCloseAccessText"   :       "Drücken Sie die Eingabetaste, um die Datei zu löschen ",
-  "FileSizeGreater"       :        "Dateien {0} übersteigen die erwartete Größe: {1}MB.",
-  "FileNameInvalid"       :       "Datei(en) {0} hat/haben ungültige Zeichen in ihrem Namen. Es werden nur alphanumerische Zeichen unterstützt",
-  "FileMimeTypeInvalid"   :       "Datei(en) {0} ist/sind nicht unterstützte(r) Dateityp(en)",
-  "InternalFormSubmissionError" :  "Beim Übermitteln des Formulars ist ein interner Fehler aufgetreten."
+  "FileCloseAccessText"   :           "Drücken Sie die Eingabetaste, um die Datei zu löschen ",
+  "FileSizeGreater"       :           "Dateien {0} übersteigen die erwartete Größe: {1}MB.",
+  "FileNameInvalid"       :           "Datei(en) {0} hat/haben ungültige Zeichen in ihrem Namen. Es werden nur alphanumerische Zeichen unterstützt",
+  "FileMimeTypeInvalid"   :           "Datei(en) {0} ist/sind nicht unterstützte(r) Dateityp(en)",
+  "InternalFormSubmissionError" :     "Beim Übermitteln des Formulars ist ein interner Fehler aufgetreten.",
```
Comment (rismehta) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/af-clientlibs/core-forms-components-runtime-all/resources/i18n/de.json:6]: Can we match the name to the validationState, it becomes easier that way

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/af-clientlibs/core-forms-components-runtime-all/resources/i18n/it.json [hunk 3]
```diff
@@ -3,5 +3,21 @@
   "FileSizeGreater"       :        "I file {0} superano le dimensioni previste: {1} MB.",
   "FileNameInvalid"       :       "I file {0} contengono caratteri non validi nel nome. Sono supportati solo caratteri alfanumerici",
   "FileMimeTypeInvalid"   :       "I file {0} non sono tipi di file supportati",
-  "InternalFormSubmissionError" :  "Errore interno durante l'invio del modulo."
+  "InternalFormSubmissionError" :  "Errore interno durante l'invio del modulo.",
+  "rangeUnderflow" :                  "",
```
Comment (rismehta) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/af-clientlibs/core-forms-components-runtime-all/resources/i18n/it.json:7]: Please add english strings for all these

File: ui.frontend/src/view/FormFieldBase.js [hunk 4]
```diff
@@ -221,9 +222,60 @@ export default class FormFieldBase extends FormField {
...
+                    ACCEPT_MISMATCH : 'acceptMismatch',
+                    FILE_SIZE_MISMATCH: 'fileSizeMismatch',
+                    UNIQUE_ITEMS_MISMATCH: 'uniqueItemsMismatch',
+                    MIN_ITEMS_MISMATCH: 'minItemsMismatch',
+                    MAX_ITEMS_MISMATCH: 'maxItemsMismatch',
+                    EXPRESSION_MISMATCH : 'expressionMismatch'
+                });
+
+                function getSnippetForConstraint(validationState) {
+                    let snippets_Array = [];
```
Comment (rismehta) [inline ui.frontend/src/view/FormFieldBase.js:246]: Please following the naming conventions here

File: ui.tests/test-module/specs/checkboxgroup/checkboxgroup.runtime.spec.js [hunk 5]
```diff
@@ -156,7 +156,7 @@ describe("Form Runtime with CheckBoxGroup Input", () => {
 
         cy.get(`#${checkBox3}`).find("input").uncheck().check(["0"]).blur().then(x => {
             cy.get(`#${checkBox5}`).find("input").uncheck().check(["1"])
-            cy.get(`#${checkBox5}`).find(".cmp-adaptiveform-checkboxgroup__errormessage").should('have.text',"There is an error in the field")
+            cy.get(`#${checkBox5}`).find(".cmp-adaptiveform-checkboxgroup__errormessage").should('have.text',"There is an error in the field.")
```
Comment (rismehta) [inline ui.tests/test-module/specs/checkboxgroup/checkboxgroup.runtime.spec.js:159]: Please add a test case to test different error messages

**Review by rismehta — CHANGES_REQUESTED** [review]
-1 for now

**Comment by s1mahanty** [comment]
Closing as approach changed, new changes followed up in the PR: https://github.com/adobe/aem-core-forms-components/pull/714

**Pull request closed without merging**